web/templates/monitor/details_processor.go.html

```diff
@@ -2,497 +2,518 @@
-          }
-          lastTableStats.set(this, stats);
-
-          return '<td>'+stats.partition.toFixed(0)+'</td>\n'+
-            '<td>'+status+'</td>\n'+
-            '<td>'+recoveryTime+'s</td>\n'+
-            '<td>'+recoveryRate.toFixed(0)+'</td>\n'+
-            '<td>'+recoveryLag+'</td>\n'+
-            '<td>'+writeRate.toFixed(0)+'</td>\n'+
-            '<td>'+writeBytes.toFixed(0)+'</td>\n';
-        };
-
-        var groupChain = _.chain(partitions.Group);
-        var partitionList = groupChain
-              .filter(function(v){return v.TableStats != null; })
-              .map(function(v, partId){
-                return _.assign(
-                  {},
-                  v.TableStats,
-                  { 'Now': new Date(v.Now),
-                    'partition' : parseInt(partId),
-                  },
-                );
-              }).values().sortBy('partition').value();
-
-        // sort all partitions
-        // update, enter and remove data
-        var d = d3.select("#tableStatistics").selectAll(".partitionbox").data(partitionList);
-        d.html(updateTableStatistics);
-        d.enter().append("tr").classed("partitionbox", true).html(updateTableStatistics);
-        d.exit().remove()
-
-        var renderInput = function(data){
-          var topic = data[0];
-          var stats = data[1];
-
-          // default values we're going to display.
-          // this object will be stored in lastInputStats to calculate rates as well
-          var values = {
-            "now":new Date(),
-            "countSum" : _.sum(stats.Count),
-            "countAvg" : _.mean(stats.Count),
-            "bytesSum" : _.sum(stats.Bytes),
-            "bytesAvg" : _.mean(stats.Bytes),
-          };
+            var writeRate = 0;
+            var writeBytes = 0;
 
+            // let's check if we have a previous dataset and set them to their
+            // correct values
+            var lastStats = lastTableStats.get(this);
 
-          var countRate = 0;
-          var countRateAvg = 0;
-          var bytesRate = 0;
-          var bytesRateAvg = 0;
+            if (lastStats) {
+              // diff to previous stats in seconds
+              var timeDiff = (stats.Now - lastStats.Now) / 1000.0;
 
-          var lastStats = lastInputStats.get(this);
+              if (status == "recovering") {
+                recoveryRate = (stats.Recovery.Offset - lastStats.Recovery.Offset) / timeDiff;
+              }
 
-          if(lastStats){
-            // diff to previous stats in seconds
-            var timeDiff = (values.now - lastStats.now) / 1000.0;
+              writeRate = (stats.Writes.Count - lastStats.Writes.Count) / timeDiff;
+              writeBytes = (stats.Writes.Bytes - lastStats.Writes.Bytes) / timeDiff;
+            }
+            lastTableStats.set(this, stats);
+
+            return '<td>' + stats.partition.toFixed(0) + '</td>\n' +
+              '<td>' + status + " (" + runMode + ')</td>\n' +
```

Comment:
the file got reformatted, so the changes are rather small. I only added the runMode to the stats for table-partition and joins (see screenshot in PR description)